src/api.proposed.x.d.ts

```diff
@@ -0,0 +1,167 @@
+         */
+        readonly webSocketProtocols?: string[];
+    }
+
+    /**
+     * Represents a Jupyter Server displayed in the list of Servers.
+     */
+    export interface JupyterServer {
+        /**
+         * Unique identifier for this server.
+         */
+        readonly id: string;
+        /**
+         * A human-readable string representing the name of the Server. This can be read and updated by the extension.
+         */
+        label: string;
+        /**
+         * Returns the connection information for this server.
+         */
+        resolveConnectionInformation(token: CancellationToken): Promise<JupyterServerConnectionInformation>;
+    }
+
+    /**
+     * Provider of Jupyter Servers.
+     */
+    export interface JupyterServerProvider {
+        /**
+         * Event fired when the list of servers changes.
+         * Note: the getJupyterServers method will not be called unless changes are detected.
+         */
+        onDidChangeServers: Event<void>;
+        /**
+         * Returns the list of servers.
+         */
+        getJupyterServers(token: CancellationToken): Promise<JupyterServer[]>;
+    }
+    /**
+     * Represents a reference to a Jupyter Server command. Provides a title which
+     * will be used to represent a command in the UI and, optionally,
+     * an array of arguments which will be passed to the command handler
+     * function when invoked.
+     */
+    export interface JupyterServerCommand {
+        /**
+         * Title of the command, like `save`.
+         */
+        title: string;
+        /**
+         * A human-readable string which is rendered less prominent in a separate line.
+         */
+        detail?: string;
+        /**
+         * The identifier of the actual command handler.
+         * @see {@link commands.registerCommand}
+         */
+        command: string;
+        /**
+         * A tooltip for the command, when represented in the UI.
+         */
+        tooltip?: string;
+        /**
+         * Arguments that the command handler should be
+         * invoked with.
+         */
+        arguments?: unknown[];
+    }
+    /**
+     * Provider of Jupyter Server Commands.
+     * Each command allows the user to perform an action.
+     * The return value of the command should be of the form Promise<JupyterServer | 'back' | undefined>
+     * The returned value have the following meaning:
+     * - JupyterServer  : The Jupyter Server object that was created
+     * - 'back'         : Go back to the previous screen
+     * - undefined|void : Do nothing
+     */
+    export interface JupyterServerCommandProvider {
+        /**
+         * Default command to be used when there are no servers. This can be read and updated by the extension.
+         * If not set, and there are not servers, then the user will be prompted to select a command from a list of commands returned by `getCommands`.
+         */
+        selected?: JupyterServerCommand;
```

Comment:
Is there a better name than `selected`? I'm finding the name confusing even after reading the docs

Also having both `selected` and `getCommands` is confusing me. Could `getCommands` handle both cases?